vissl/engines/slurm_evaluator.py

```diff
@@ -0,0 +1,506 @@
+                for iteration in evaluate_iterations:
+                    training_checkpoint = f"model_iteration{iteration}"
+                    self._set_initial_benchmark_result(
+                        benchmark,
+                        training_checkpoint,
+                        evaluation_name,
+                        evaluation_configs,
+                    )
+
+        return evaluation_configs
+
+    def _num_training_iterations(self):
+        cfg = self.training_config
+
+        num_iterations = None
+        if self.max_training_iterations:
+            num_iterations = self.max_training_iterations
+        else:
+            # TODO: This can potentially be improved.
+            train_cfg = cfg.DATA.TRAIN
+            global_batch_size = (
+                train_cfg.BATCHSIZE_PER_REPLICA
+                * cfg.DISTRIBUTED.NUM_NODES
+                * cfg.DISTRIBUTED.NUM_PROC_PER_NODE
+            )
+            iterations_per_epoch = train_cfg.DATA_LIMIT // global_batch_size
+            num_iterations = iterations_per_epoch * cfg.OPTIMIZER.num_epochs
+
+        return num_iterations
+
+    def _load_evaluation_results_checkpoint(self):
+        default_checkpoint = os.path.join(
+            self.evaluation_dir(), "evaluation_metrics.json"
+        )
+        checkpoint_file = (
+            default_checkpoint if self.autoload_slurm_evaluator_checkpoint else self.slurm_evaluator_checkpoint
+        )
+
+        evaluation_config = load_file(checkpoint_file)
+
+        print(f"Loaded evaluation results checkpoint from: { checkpoint_file }")
+
+        return evaluation_config
+
+    def _set_initial_benchmark_result(
+        self, benchmark, training_checkpoint, evaluation_name, evaluation_configs
+    ):
+        """
+        Generates evaluation configs in order to evaluate the final output of the
+        pretraining model specified in 'training_config'.
+        """
+        log_dir = self._evaluation_log_dir(training_checkpoint, evaluation_name)
+        checkpoint_dir = self._evaluation_checkpoint_dir(
+            training_checkpoint, evaluation_name
+        )
+
+        evaluation_configs[training_checkpoint] = (
+            evaluation_configs.get(training_checkpoint) or []
+        )
+
+        # Add benchmark result information
+        benchmark_result = {}
+        benchmark_result["evaluation_name"] = evaluation_name
+        benchmark_result["job_id"] = None
+        benchmark_result["slurm_log_dir"] = log_dir
+        benchmark_result["checkpoint_dir"] = checkpoint_dir
+        benchmark_result["metrics"] = None
+        benchmark_result["slurm_state"] = None
+        benchmark_result["config_files"] = benchmark["config_files"].copy()
+
+        # Hydra config information
+        weights_init_path = os.path.join(
+            self.training_config.CHECKPOINT.DIR, f"{ training_checkpoint }.torch"
+        )
+        config_files = benchmark_result["config_files"]
+        config_files.insert(
```

Comment:
I think you could `append` instead of `insert(1, )` unless you are worried that these options are already in the `"config_files"` configuration?

Reply:
Correct, I want those already in config_files to overwrite these options. E.g. specifying specific log directory.